### PR TITLE
Add VeChain platform [WIP]

### PR DESCRIPTION
**Pull request by kolya182**
Closes https://github.com/TrustWallet/blockatlas/issues/52

TODO:
- [x] Add vthor transactions

### CODE REVIEW

File: platform/vechain/api.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+func getTransactions(c *gin.Context) {
+	addr := strings.ToLower(c.Param("address"))
+	txs, _ := client.GetAddressTransactions(addr)
+
+	var receiptsMap = make(map[string]TxReceipt)
+	receiptsChan := make(chan TxReceipt, len(txs))
+
+	for _, t := range txs {
+		wg.Add(1)
+		go client.GetTransactionReceipt(receiptsChan, t.Meta.TxID)
```
Comment (terorie) [inline platform/vechain/api.go]: This could open a lot of connections, consider HTTP/1.1 pipelining or HTTP/2 multiplexing as a more efficient alternative

Comment (kolya182) [reply platform/vechain/api.go]: This code was missed during rebase, added close channel here https://github.com/TrustWallet/blockatlas/pull/70/commits/9202343889007b69ca4b882a786aae6601794960. It's great idea to use `fasthttp` lib and pipelining, but for let's complete all planned chains and next step we can use your proposed improvements.

Comment (terorie) [reply platform/vechain/api.go]: Agreed, let's postpone it.

**Comment by kolya182** [comment]
I decided use VeChain explorer API instead of `thor` RPC for couple of reasons
1. Consistency and accuracy
2. Speed

IMG_5087
IMG_5086

**Comment by vikmeup** [comment]
Very nice!

File: platform/vechain/api.go [hunk 2]
```diff
@@ -91,44 +91,48 @@ func GetAddressTransactions(address string, token string) []models.Tx {
...
-			value = hexaToIntegerString(output.Transfers[0].Amount)
+	tx = models.Tx{
+		ID:       tr.Meta.TxID,
+		Coin:     coin.VET,
+		From:     sender,
+		To:       recipient,
+		Date:     timestamp,
+		Type:     transferType,
+		Block:    tr.Meta.BlockNumber,
+		Sequence: uint64(timestamp),
```
Comment (terorie) [inline platform/vechain/api.go]: iirc, we chose to omit `Sequence` for coins that don't have tx sequences/nonces, so you can remove this

File: platform/vechain/api_test.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+	
+		if !bytes.Equal(actual, expected) {
+			println(string(actual))
+			println(string(expected))
+			t.Error("Transactions not equal")
+		}
+	}
+}
+
+// func Unmarshal(t testing.T, input string, i *interface{}) {
```
Comment (terorie) [inline platform/vechain/api_test.go]: unused comment

File: test/main.go [hunk 4]
```diff
@@ -40,6 +40,7 @@ var addresses = map[string]Entry{
 	"thundertoken":     {coin.TT,  "0x0ad80a408eac4f17ba0a9de8a12d8736f60700c3"},
 	"icon":             {coin.ICX, "hxee691e7bccc4eb11fee922896e9f51490e62b12e"},
 	"tron":             {coin.TRX, "TMuA6YqfCeX8EhbfYEg5y7S4DqzSJireY9"},
+	"vechain":          {coin.VET,  "0xB5e883349e68aB59307d1604555AC890fAC47128"},
```
Comment (terorie) [inline test/main.go]: Whitespace before address :eyes:

**Pull request merged**